src/WebJobs.Extensions.DurableTask/Listener/TaskHttpActivityShim.cs

```diff
@@ -0,0 +1,112 @@
+    {
+        private readonly HttpClient httpClient;
+        private readonly DurableTaskExtension config;
+
+        public TaskHttpActivityShim(
+            DurableTaskExtension config,
+            HttpClient httpClientFactory)
+        {
+            this.config = config ?? throw new ArgumentNullException(nameof(config));
+            this.httpClient = httpClientFactory;
+        }
+
+        public override string Run(TaskContext context, string input)
+        {
+            // This won't get called as long as we've implemented RunAsync.
+            throw new NotImplementedException();
+        }
+
+        public async override Task<string> RunAsync(TaskContext context, string rawInput)
+        {
+            HttpRequestMessage requestMessage = await CreateHttpRequestMessageAsync(rawInput);
+            HttpResponseMessage response = await this.httpClient.SendAsync(requestMessage);
+            DurableHttpResponse durableHttpResponse = await CreateDurableHttpResponseAsync(response);
+
+            string serializedOutput = MessagePayloadDataConverter.HttpConverter.Serialize(durableHttpResponse, true);
+            return serializedOutput;
+        }
+
+        private static async Task<DurableHttpResponse> CreateDurableHttpResponseAsync(HttpResponseMessage response)
+        {
+            DurableHttpResponse durableHttpResponse = new DurableHttpResponse(response.StatusCode);
+            durableHttpResponse.Headers = CreateStringValuesHeaderDictionary(response.Headers);
+            durableHttpResponse.Content = await response.Content.ReadAsStringAsync();
+            return durableHttpResponse;
+        }
+
+        private static IDictionary<string, StringValues> CreateStringValuesHeaderDictionary(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
+        {
+            IDictionary<string, StringValues> newHeaders = new Dictionary<string, StringValues>();
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    foreach (var headerValue in header.Value)
+                    {
+                        if (newHeaders.ContainsKey(header.Key))
+                        {
+                            StringValues values = StringValues.Concat(headerValue, newHeaders[header.Key]);
+                            newHeaders[header.Key] = values;
+                        }
+                        else
+                        {
+                            StringValues stringValues = new StringValues(headerValue);
+                            newHeaders.Add(header.Key, stringValues);
+                        }
+                    }
+                }
+            }
+
+            return newHeaders;
+        }
+
+        private static async Task<HttpRequestMessage> CreateHttpRequestMessageAsync(string rawInput)
+        {
+            JsonSerializerSettings serializerSettings = new JsonSerializerSettings();
```

Comment:
In general, if you are using some fixed serializer settings here, it is best to at least make it a class or static variable, so you aren't constantly recreating the same `JsonSerializerSettings` object every time you call the method.